parser/src/ast/transition_constraints.rs

```diff
@@ -49,3 +43,28 @@ pub enum TransitionExpr {
     Mul(Box<TransitionExpr>, Box<TransitionExpr>),
     Exp(Box<TransitionExpr>, u64),
 }
+
+#[derive(Debug, PartialEq)]
+pub struct TransitionVariable {
+    name: Identifier,
+    value: TransitionVariableType,
+}
+
+impl TransitionVariable {
+    pub fn new(name: Identifier, value: TransitionVariableType) -> Self {
+        Self { name, value }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum TransitionVariableType {
+    Scalar(TransitionExpr),
+    Vector(Vec<TransitionExpr>),
+    Matrix(Vec<Vec<TransitionExpr>>),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum TransitionStmt {
+    Constraint(TransitionConstraint),
+    Variable(TransitionVariable),
+}
```

Comment:
nit: I would probably put this towards the top of the file as this is the top level struct.